src/annotator/hypothesis-injector.js

```diff
@@ -85,13 +85,13 @@ export class HypothesisInjector {
 }
 
 /**
- * Check if the Hypothesis client has already been injected into an iframe
+ * Check if the client was added to a frame by {@link injectHypothesis}.
  *
  * @param {HTMLIFrameElement} iframe
  */
 function hasHypothesis(iframe) {
-  const iframeWindow = /** @type {Window} */ (iframe.contentWindow);
-  return '__hypothesis' in iframeWindow;
+  const iframeDocument = /** @type {Document} */ (iframe.contentDocument);
```

Comment:
This will raise an error if accessing a cross-origin frame.

Reply:
In the context `hasHypothesis` is used, the `injectClient` method, it is specified that a prerequisite is for the frame to be same-origin. An option would be to simply mention this requirement in the docs.